File: TD/TD7/Code/P1.cpp
/**
 * Remplir le tableau avec le résultat du produit de la ligne par colonne
 */
#include <iostream>

const int MAX = 100;

/* Procédure */

void tableMultip(int tab[MAX][MAX], int n)
{
  for (int i = 0; i < n; ++i)
    for (int j = 0; j < i + 1; ++j)
      tab[i][j] = i * j;
}

/* Utilisation */

int main()
{
  int tab[MAX][MAX];
  
  tableMultip(tab, 5);
  
  // Afichage de ce tableau
  for (int i = 0; i < 5; ++i) {
    for (int j = 0; j < i + 1; ++j)
      std::cout << tab[i][j] << " " << std::flush;
    std::cout << std::endl;
  }
  
  return 0;
}

File: TD/TD7/Code/Python/P1.py
#!/usr/bin/python
# Remplir le tableau avec le résultat du produit de la ligne par colonne


def table_multip(tab):
    for i in range(len(tab)):
        for j in range(i + 1):
            tab[i][j] = i * j


tab = [[0 for i in range(6)] for j in range(6)]

table_multip(tab)

# Affichage de tableau
for i1 in range(len(tab)):
    for j1 in range(i1 + 1):
        print(tab[i1][j1], '', sep='\t', end='')
    print('')
